### ISSUE TITLE
Full chat history scan even if data range set when exporting chat history 

### ISSUE THREAD

**Issue by lingo9** [telegramdesktop/tdesktop#25139]
### Steps to reproduce

1. Select random group
2. Choose export chat history, and select a date range, e.g. the last three days
3. Wait and see the client scaning for millions of messages

### Expected behaviour

Only small number of messages are scanned when selecting short date ranges.

### Actual behaviour

See 3.

### Operating system

Windows 10

### Version of Telegram Desktop

4.2 x64 portable

### Installation source

Static binary from official website

**Comment by runo280** [comment]
This is happen even for channels.

**Operation system**

Linux Mint

**Version of Telegram Desktop**

4.5.2

**Installation source**

Github release

**Comment by andreymal** [comment]
**@github-actions** shut up please, lack of activity doesn't mean that this issue should be closed

**Comment by andreymal** [comment]
**@github-actions** \*angry noises\*

**Comment by Aokromes** [comment]
not really a bug.

**Comment by andreymal** [comment]
**@github-actions** ъуъ

**Comment by andreymal** [comment]
**@github-actions** beep boop

**Comment by OzzyCallooh** [comment]
Yep, this is still broken. I thought this may have been an issue with the topics feature, but no, it's definitely still a problem in 6.1.1

**Comment by andreymal** [comment]
**@github-actions** this issue was inactive for a long time, we therefore assume that Telegram developers have lost interest to resolve the problem

**Comment by ykozatskyi** [comment]
Same issue here. When I do a chat export and choose a date range - lets say today only - the client goes to FULL scan entire chat.

**Comment by AnLuoRidge** [comment]
I've been having this issue for at least 2 years. Didn't realise it was raised in 2022...
